packages/core/src/agent/Agent.ts

```diff
@@ -233,8 +235,8 @@ export class Agent {
       const mediationConnection = await this.getMediationConnection(mediatorConnectionsInvite)
       await this.mediationRecipient.provision(mediationConnection)
     }
-
-    await this.mediationRecipient.initialize()
+    if (mediationRole === MediationRole.Mediator) await this.mediator.initialize()
+    if (mediationRole === MediationRole.Recipient) await this.mediationRecipient.initialize()
```

Comment:
Should not we also check if `mediationRole === MediationRole.Recipient` before we try to use `mediatorConnectionsInvite`?